lib/s3middleware/objectRestore.ts

```diff
@@ -0,0 +1,141 @@
+import { parseString } from 'xml2js';
+import errors, { ArsenalError } from '../errors';
+import * as werelogs from 'werelogs';
+
+/*
+    Format of xml request:
+    <RestoreRequest>
+      <Days>integer</Days>
+      <Tier>Standard|Bulk|Expedited</RetainUntilDate>
+    </RestoreRequest>
+*/
+
+/**
+ * validateDays - validate restore days
+ * @param days - parsed days string
+ * @return - contains days or error
```

Comment:
I would not return something or error, especially since this is an object with errors defined or days defined and not X|Y.
I would consider either renaming the function to something like `getDays` that throws an exception on a bad value.
Or maybe we find another style that works but I wouldn't return two different things for two different reasons.

Reply:
agree with you, I changed the structure to have one function only for validating, returning errors on failure or undefined on success